src/email.cc

```diff
@@ -234,15 +234,15 @@ Smtp::msg2email (zmsg_t **msg_p) const
 
     char *to = zmsg_popstr (msg);
     char *subject = zmsg_popstr (msg);
-    char *body = zmsg_popstr (msg);
+    std::string body = getIpAddr();
+    body += zmsg_popstr (msg);
```

Comment:
Memory leak, the result of zmsg_popstr must be freed with zstr_free or, preferably, be put inside a ZstrGuard.